src/go/util/httppattern/uri_template_impl.go

```diff
@@ -0,0 +1,454 @@
+// Copyright 2019 Google LLC
```

Comment:
Splitting into impl is c++ style, not golang style. We don't split the public interface from the implementation like this. How about change this file as `uri_template_parser.go`, and then move the 'serialize`  to `http_template.go`?